presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -89,15 +100,50 @@ private PinotExpression handleLogicalBinary(
         }
         List<RowExpression> arguments = call.getArguments();
         if (arguments.size() == 2) {
-            return derived(format(
-                    "(%s %s %s)",
-                    arguments.get(0).accept(this, context).getDefinition(),
-                    operator,
-                    arguments.get(1).accept(this, context).getDefinition()));
+            String def0 = arguments.get(0).accept(this, context).getDefinition();
```

Comment:
I think I would prefer handleLogicalBinary to check for the presence of date/time etc and then delegate to a helper method instead of inlining the logic here. Here is what I mean in code:

```
if (isDateTimeExpression(arguments)) {
  return handleDateTimeBinaryExpression(operator, call, context) // or pass it arguments, context ..  whatever
}
// Continue normal handleLogicalBinary flow here
```

Reply:
will do